examples/fishnet_api/main.py

```diff
@@ -164,14 +227,16 @@ async def upload_dataset(dataset: UploadDatasetRequest) -> Dataset:
     if dataset.ownsAllTimeseries:
         # check if _really_ owns all timeseries
         timeseries = await Timeseries.fetch(dataset.timeseriesIDs)
-        dataset.ownsAllTimeseries = all([ts.owner == dataset.owner for ts in timeseries])
+        dataset.ownsAllTimeseries = all(
+            [ts.owner == dataset.owner for ts in timeseries]
+        )
     if dataset.id_hash is not None:
         # update existing dataset
-        resp = (await Dataset.fetch(dataset.id_hash))
+        resp = await Dataset.fetch(dataset.id_hash)
         old_dataset = resp[0] if resp else None
         if old_dataset is not None:
             if old_dataset.owner != dataset.owner:
-                raise Exception("Cannot overwrite dataset that is not owned by you")
+                raise HTTPException(status_code=404, detail="Cannot overwrite dataset that is not owned by you")
```

Comment:
Same here, should be "Forbidden"
```suggestion
                raise HTTPException(status_code=403, detail="Cannot overwrite dataset that is not owned by you")
```